src/ResourceManager/Network/Commands.Network/NetworkInterface/TapConfiguration/NetworkInterfaceTapConfigBase.cs

```diff
@@ -0,0 +1,72 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    public abstract class NetworkInterfaceTapConfigBase : NetworkInterfaceBaseCmdlet
```

Comment:
Why are you deriving from NetworkInterfaceBaseCmdlet ?

Reply:
Please read the tapconfig cmdlet implementation, I’m using one of the APIs